File: EticaretJava/app/src/main/java/com/example/eticaretjava/model/Product.java
package com.example.eticaretjava.model;


import com.google.gson.annotations.SerializedName;

public class Product {

    public static class ProductListDto {
        @SerializedName("id")
        public int id;

        @SerializedName("name")
        public String name;

        @SerializedName("price")
        public double price;

        @SerializedName("discount_percent")
        public Double discountPercent;

        @SerializedName("image_url")
        public String imageUrl;

        @SerializedName("stock_qty")
        public int stockQty;

        @SerializedName("is_active")
        public int isActive;
    }

    public static class ProductDto {
        @SerializedName("id")
        public int id;

        @SerializedName("name")
        public String name;

        @SerializedName("slug")
        public String slug;

        @SerializedName("sku")
        public String sku;

        @SerializedName("price")
        public double price;

        @SerializedName("discount_percent")
        public Double discountPercent;

        @SerializedName("image_url")
        public String imageUrl;

        @SerializedName("stock_qty")
        public int stockQty;

        @SerializedName("is_active")
        public int isActive;

        @SerializedName("created_at")
        public String createdAt;

        @SerializedName("updated_at")
        public String updatedAt;
    }
}

File: Eticaretjetpack/app/src/main/java/com/example/eticaretjetpack/model/Product.kt
package com.example.eticaretjetpack.model

import com.google.gson.annotations.SerializedName

// Ürün listesi için sade DTO
data class ProductListDto(
    @SerializedName("id")
    val id: Int,

    @SerializedName("name")
    val name: String,

    @SerializedName("price")
    val price: Double,

    @SerializedName("discount_percent")
    val discountPercent: Double?,

    @SerializedName("image_url")
    val imageUrl: String?,

    @SerializedName("stock_qty")
    val stockQty: Int,

    @SerializedName("is_active")
    val isActive: Int
)

// Ürün detay DTO
data class ProductDto(
    @SerializedName("id")
    val id: Int,

    @SerializedName("name")
    val name: String,

    @SerializedName("slug")
    val slug: String,

    @SerializedName("sku")
    val sku: String,

    @SerializedName("price")
    val price: Double,

    @SerializedName("discount_percent")
    val discountPercent: Double?,

    @SerializedName("image_url")
    val imageUrl: String?,

    @SerializedName("stock_qty")
    val stockQty: Int,

    @SerializedName("is_active")
    val isActive: Int,

    @SerializedName("created_at")
    val createdAt: String,

    @SerializedName("updated_at")
    val updatedAt: String?
)
